web/wp-content/themes/lfevents/template-parts/upcoming-events-loop.php

```diff
@@ -0,0 +1,125 @@
+					$date_range    = jb_verbose_date_range( $dt_date_start, $dt_date_end );
+				}
+
+				$register_url = get_post_meta( $post->ID, 'lfes_cta_register_url', true );
+				$speak_url    = get_post_meta( $post->ID, 'lfes_cta_speak_url', true );
+				$sponsor_url  = get_post_meta( $post->ID, 'lfes_cta_sponsor_url', true );
+				$schedule_url = get_post_meta( $post->ID, 'lfes_cta_schedule_url', true );
+				$description  = get_post_meta( $post->ID, 'lfes_description', true );
+				?>
+
+				<div id="post-<?php the_ID(); ?>" class="cell medium-6">
+
+					<h4 class="medium-margin-right small-margin-bottom line-height-tight">
+						<a class="unstyled-link" href="<?php echo esc_html( lfe_get_event_url( $post->ID ) ); ?>">
+							<strong><?php echo esc_html( get_the_title( $post->ID ) ); ?></strong>
+						</a>
+					</h4>
+
+					<p class="event-meta text-small small-margin-bottom">
+						<span class="date small-margin-right display-inline-block">
+							<?php get_template_part( 'template-parts/svg/calendar' ); ?>
+							<?php echo esc_html( $date_range ); ?>
+						</span>
+
+						<span class="country display-inline-block">
+							<?php get_template_part( 'template-parts/svg/map-marker' ); ?>
+							<?php
+							$country = wp_get_post_terms( $post->ID, 'lfevent-country' );
+							if ( $country ) {
+								$country = $country[0]->name;
+								$city    = get_post_meta( $post->ID, 'lfes_city', true );
+								if ( $city ) {
+									$city .= ', ';
+								}
+								echo esc_html( $city ) . esc_html( $country );
+							}
+							?>
+						</span>
+					</p>
+
+					<p class="text-small small-margin-bottom">
+						<?php
+						echo esc_html( $description );
+						?>
+					</p>
+
+					<p class="homepage--call-to-action">
+						<?php
+						if ( $register_url ) {
+							echo '<a aria-label="Register for ' . esc_html( get_the_title( $post->ID ) ) . '" href="' . esc_url( $register_url ) . '" >Register</a>';
+						}
+
+						if ( $speak_url ) {
+							echo '<a aria-label="Speak at ' . esc_html( get_the_title( $post->ID ) ) . '" href="' . esc_url( $speak_url ) . '">Speak</a>';
+						}
+
+						if ( $sponsor_url ) {
+							echo '<a aria-label="Sponsor ' . esc_html( get_the_title( $post->ID ) ) . '" href="' . esc_url( $sponsor_url ) . '">Sponsor</a>';
+						}
+
+						if ( $schedule_url ) {
+							echo '<a aria-label="View schedule for ' . esc_html( get_the_title( $post->ID ) ) . '" href="' . esc_url( $schedule_url ) . '">Schedule</a>';
+						}
+
+						if ( ! $register_url && ! $speak_url && ! $sponsor_url && ! $schedule_url ) {
+							echo '<a aria-label="Learn more about ' . esc_html( get_the_title( $post->ID ) ) . '" href="' . esc_html( lfe_get_event_url( $post->ID ) ) . '">Learn more</a>';
+						}
+						?>
+					</p>
+
+				</div>
+
+				<?php
+			}
+			wp_reset_postdata();
+		} else {
+			echo "Sorry there aren't any events currently scheduled.";
```

Comment:
Why isn't this currently showing [here](https://pr-608-lfasiallcci.pantheonsite.io/)?  And can you make it "Sorry there aren't any upcoming events scheduled."  I will get a Chinese translation.

Reply:
From what I understand the query always returns results (all events) but then filters passed/hidden events as part of the while loop. I've added a simple counter, and if it's 0 results then it displays the no events message.